src/game-loop/src/game-loop/GameLoopPlayingState.cpp

```diff
@@ -152,6 +160,29 @@ void GameLoopPlayingState::enter(GameLoop& game_loop)
     populator::generate_inventory_items(_main_dude);
 
     game_loop._level_summary_tracker->entered_new_level();
+
+    // Subscribe HUD on main-dude's wallet:
+
+    auto& item_carrier = registry.get<ItemCarrierComponent>(_main_dude);
```

Comment:
There's already a helper method for getting entity of `ItemType` from `ItemCarrierComponent`